### ISSUE TITLE
Do not render building names if they are also a power feature

### ISSUE THREAD

**Issue by SLMapper** [gravitystorm/openstreetmap-carto#4739]
### Expected behavior
Names of street cabinets, transformer towers ~~and power poles~~ (and others power features?) should not be shown on the map. 
_(update: seems for power poles it is not displayed - at least the example I was looking at)_

Why?
- Because they are not what most users are looking for at a map
- They often have the name of a nearby location, street or even the family living close. But they *are not* the street, the location or family home, which causes confusion or at least disturbs

### Actual behavior
They are visible 

### Screenshots with links illustrating the problem
https://www.openstreetmap.org/way/770877326#map=19/48.55499/8.96905
![IMG_20221129_133511](https://user-images.githubusercontent.com/104768113/204530798-e0a6dfe0-d4f7-4dfc-a3f1-c06ceb1b439c.jpg)

https://www.openstreetmap.org/query?lat=48.56633&lon=8.94672
![IMG_20221129_145303](https://user-images.githubusercontent.com/104768113/204548034-8940cfd8-bcda-4f07-a78c-3c98708d92d2.jpg)

**Comment by SLMapper** [comment]
Yes, street cabinet and transformer tower are considerer a `building`. But they get their names from the electricity company or the state and are also only relevant for them. 

So if somebody tells you: "let's meet next to Hardtwald" would you think of the transformer tower or the forrest?

**Comment by imagico** [comment]
I don't want to discuss the specifics of this case here because that would get us deeply into the semantics of OSM tagging and the philosophic discussion of what a proper name is and i lack the specific knowledge of that area to assess if what is tagged there as `name=*` is (a) a proper name and (b) applies to the building and not something else.  Keep in mind though that not every identifier string that is labeled on a feature or that is used by someone to address a feature is necessarily a proper name of that feature.

We expect the `name=*` tag of a feature tagged `building=*` to be the proper name of the building as a structure (think of things like 'Empire State Building' or 'Bendlerblock').  If what is tagged as `name=*` in your examples is not such a name it might be worth considering to either change it to a different tag (like `ref` or some `addr:*` tag) or to separate the building from its function (like you'd do in case of a shop in a building where the name of the building and the name of the shop differ).

**Comment by SLMapper** [comment]
Thank you for mentioning the problem of misusing the name key. Indeed I also try to fix this whereever I see it, e.g. a parking with `name=Parking`.

Separating the feature from the building does imho not make much sense in this case as
- the building only exists for this single purpose
- I have not seen anybody mapping it separately so far
- also on the [wiki pages](https://wiki.openstreetmap.org/wiki/Tag:building%3Dtransformer_tower) of `building=transformer_tower` and `building=service` it says
  > Useful combination
    -- power=substation
    -- substation=minor_distribution

So I think the building and the substation can be thought of as one thing.

Regardin usage of `name`:
- (at least in the cases I know) it is written to the outside of the building (see picture below)
- and many people seem to think this is the name:
  From some **sample** overpass queries in Germany I found
  - ~3000 having `name` not `ref`
  - ~1500 having `ref` but not `name`
  - ~300 having `ref` **and** `name` and there `name` often is in the same "location style" I am refering to. Examples: [1](https://www.openstreetmap.org/way/121383854), [2](https://www.openstreetmap.org/way/240066597), [3](http://www.openstreetmap.org/way/802153310).

[picture TODO]

Therefore I think we should
- accept the usage of `name` to be valid
- hide the name for these on the general purpose map osm-carto

Having it displayed on a specific map for power features however would absolutely make sense.

**Comment by imagico** [comment]
Well - as said i am not going to discuss what is correct mapping and tagging here.  It is evident that ambiguous tagging like this  exists a lot in the database but in my experience mappers broadly agree that it is ambiguous and that no consensus exists how to resolve that ambiguity.  Should the OSM community at large ever develop consensus on rules how secondary tags (like `name=*`) are to be assigned to primary tags (like `building=*`, `power=*`) in case of features with multiple primary tags, we can reconsider this matter.  Currently consensus is that such ambiguities are to be resolved by having separate features for the different concepts (like in this case the building and the power infrastructure inside the building or the mentioned shop with a name different from the building it occupies).  

Note, however, that in this case separating the building from its function would not universally avoid a label being drawn because we do label `power=substation` with a name label as well, just with a different design:

https://www.openstreetmap.org/node/8150119764
https://www.openstreetmap.org/node/389916247
https://www.openstreetmap.org/node/8602977860
https://www.openstreetmap.org/relation/8260363
https://www.openstreetmap.org/way/458217802
https://www.openstreetmap.org/way/309200267
https://www.openstreetmap.org/way/365786756
https://www.openstreetmap.org/way/100962927
https://www.openstreetmap.org/node/5516682025
https://www.openstreetmap.org/way/221914642

I would support reconsidering our decision to render name labels on power=substation` in light of the inconsistent use of the name tag on those in many countries (see examples linked to).

This would, however, not change the specific cases you use as example here so that would be kind of a separate issue.

**Comment by ZeLonewolf** [comment]
I think that rendering a name tag on a properly-tagged building, even if it is part of power infrastructure, is perfectly fine. I would propose to close this as won't-fix.
